rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1828,6 +1828,535 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
+		then extract [ item ->  value ]
+
+	alias beforeEconomicterms: businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+	alias afterEconomicTerms: FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+
+	set is_event:
+		businessEvent -> after -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> interestRatePayout exists and
+		afterEconomicTerms -> payout <> beforeEconomicterms -> payout and
+		beforeTradePurchasePrice exists and
+		afterTradePurchasePrice exists and
+		afterTradePurchasePrice  = beforeTradePurchasePrice and
+		beforeTradeCollateralQuantity <> afterTradeCollateralQuantity and
+		beforeEconomicterms exists and
+		afterEconomicTerms exists and
+		afterEconomicTerms -> terminationDate = beforeEconomicterms -> terminationDate
+		
+
+func Create_SubstitutionPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
+	inputs:
+		tradeState TradeState (1..1) <"The original trade to be for substitution of collateral.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new trade with new collateral.">
+		newCollateralPortfolio CollateralPortfolio(1..1) <"New collateral portfolio to subtitute for the original collateral.">
+		priceQuantity PriceQuantity (1..*) <"The price and quantity of the substituted product.">
+	output: 
+		instruction PrimitiveInstruction (1..1)
+
+	condition ContractualProduct: <"Only a security finance contractual product can substitute collateral.">
+		tradeState -> trade -> tradableProduct -> product -> contractualProduct exists
+
+        set instruction: <"Sets the second part of the split to be a new contract with a new asset payout.">
+                PrimitiveInstruction {
+                    quantityChange: QuantityChangeInstruction {
+                            change: priceQuantity,
+                            direction: QuantityChangeDirectionEnum -> Replace,
+                            lotIdentifier: empty
+                        },
+                    termsChange: Create_SubstitutionInstruction(
+                                tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+                               effectiveDate,
+                               newCollateralPortfolio
+                            ),
+                    ...
+                }
+
+func Create_SubstitutionInstruction:<"Creates the terms change instruction that updates the payout with the new subsitution payout.">
+	inputs:
+		contractualProduct ContractualProduct (1..1) <"The orginal contractual product to be used as the basis of the new trade.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
```

Comment:
We're no longer closing the original trade so you may want to update the comment